ui/page/root/home_page.go

```diff
@@ -0,0 +1,357 @@
+	return layout.Stack{}.Layout(gtx,
+		layout.Expanded(func(gtx C) D {
+			return cryptomaterial.LinearLayout{
+				Width:       cryptomaterial.MatchParent,
+				Height:      cryptomaterial.MatchParent,
+				Orientation: layout.Vertical,
+			}.Layout(gtx,
+				layout.Rigid(hp.LayoutTopBar),
+				layout.Rigid(func(gtx C) D {
+					return layout.Inset{
+						Left: values.MarginPadding20,
+					}.Layout(gtx, func(gtx C) D {
+						return hp.navigationTab.Layout(gtx, navigationTabTitles)
+					})
+				}),
+				layout.Rigid(hp.Theme.Separator().Layout),
+				layout.Flexed(1, func(gtx C) D {
+					return layout.Inset{Top: values.MarginPadding16}.Layout(gtx, hp.CurrentPage().Layout)
+				}),
+			)
+		}),
+	)
+}
+
+func (hp *HomePage) layoutMobile(gtx C) D {
+	return layout.Stack{}.Layout(gtx,
+		layout.Expanded(func(gtx C) D {
+			return cryptomaterial.LinearLayout{
+				Width:       cryptomaterial.MatchParent,
+				Height:      cryptomaterial.MatchParent,
+				Orientation: layout.Vertical,
+			}.Layout(gtx,
+				layout.Rigid(hp.LayoutTopBar),
+				layout.Rigid(func(gtx C) D {
+					return layout.Inset{
+						Left: values.MarginPadding20,
+					}.Layout(gtx, func(gtx C) D {
+						return hp.navigationTab.Layout(gtx, navigationTabTitles)
+					})
+				}),
+				layout.Rigid(hp.Theme.Separator().Layout),
+				layout.Flexed(1, func(gtx C) D {
+					return layout.Inset{Top: values.MarginPadding16}.Layout(gtx, func(gtx C) D {
+						return hp.CurrentPage().Layout(gtx)
+					})
+				}),
+			)
+		}),
+	)
+}
+
+func (hp *HomePage) LayoutTopBar(gtx C) D {
+	v := values.MarginPadding20
+	return cryptomaterial.LinearLayout{
+		Width:       cryptomaterial.MatchParent,
+		Height:      cryptomaterial.WrapContent,
+		Orientation: layout.Horizontal,
+		Alignment:   layout.Middle,
+		Padding: layout.Inset{
+			Right:  v,
+			Left:   v,
+			Top:    values.MarginPadding10,
+			Bottom: v,
+		},
+	}.Layout(gtx,
+		layout.Rigid(hp.totalBalanceLayout),
+		layout.Rigid(hp.notificationSettingsLayout),
+	)
+}
+
+func (hp *HomePage) totalBalanceLayout(gtx C) D {
+	return layout.W.Layout(gtx, func(gtx C) D {
+		return cryptomaterial.LinearLayout{
+			Width:       cryptomaterial.WrapContent,
+			Height:      cryptomaterial.WrapContent,
+			Orientation: layout.Vertical,
+			// Alignment:   layout.Middle,
+		}.Layout(gtx,
+			layout.Rigid(hp.totalBalanceTextAndIconButtonLayout),
+			layout.Rigid(hp.balanceLayout),
+		)
+	})
+}
+
+func (hp *HomePage) balanceLayout(gtx C) D {
+	return layout.E.Layout(gtx, func(gtx C) D {
+		return layout.Flex{}.Layout(gtx,
+			layout.Rigid(hp.LayoutUSDBalance),
+			layout.Rigid(func(gtx C) D {
+				icon := hp.Theme.Icons.RevealIcon
+				if hp.isBalanceHidden {
+					icon = hp.Theme.Icons.ConcealIcon
+				}
+				return layout.Inset{Top: values.MarginPadding5}.Layout(gtx, func(gtx C) D {
+					return hp.hideBalanceButton.Layout(gtx, icon.Layout24dp)
+				})
+			}),
+		)
+	})
+}
+
+// TODO: use real values
+func (hp *HomePage) LayoutUSDBalance(gtx C) D {
```

Comment:
IMO this feature shouldn't exist because of its reliance on the exchange rate that are not standard.

Reply:
This is based on the current UI designs shared.